app/assets/javascripts/my_modules.js

```diff
@@ -226,34 +226,37 @@ function bindEditTagsAjax() {
     });
 }
 
-// Sets callback for completing/uncompleting task
-function applyTaskCompletedCallBack() {
-  $("[data-action='complete-task'], [data-action='uncomplete-task']")
-    .on('click', function() {
-      var button = $(this);
-      $.ajax({
-        url: button.data('link-url'),
-        type: 'POST',
-        dataType: 'json',
-        success: function(data) {
-          if (data.completed === true) {
-            button.attr('data-action', 'uncomplete-task');
-            button.find('.btn')
-              .removeClass('btn-primary').addClass('btn-default');
-          } else {
-            button.attr('data-action', 'complete-task');
-            button.find('.btn')
-              .removeClass('btn-default').addClass('btn-primary');
-          }
-          $('#dueDateContainer').html(data.module_header_due_date);
-          initDueDatePicker();
-          $('.task-state-label').html(data.module_state_label);
-          button.find('button').replaceWith(data.new_btn);
-        },
-        error: function() {
+function applyTaskStatusChangeCallBack() {
+  $('#dropdownTaskFlowList > li[data-state-id]').on('click', function() {
+    var list = $('#dropdownTaskFlowList');
+    var item = $(this);
+    var conteiner = list.closest('.task-flows');
+    console.log('click')
+    $.ajax({
+      url: list.data('link-url'),
+      type: 'PATCH',
+      contentType: 'application/json',
+      data: JSON.stringify({ status_id: item.data('state-id') }),
+      success: function(data) {
+        console.log(data.content);
+        conteiner.html(data.content)
+        HelperModule.flashAlertMsg('Greate status will be updated', 'success');
```

Comment:
Put it in locales.

Reply:
> Nice job 🎉
> I have some comments.

Its WIP, not finished yet, but anyway, thanks for comments ;)